src/main/java/brobot/Storage.java

```diff
@@ -0,0 +1,102 @@
+package brobot;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import brobot.task.Deadline;
+import brobot.task.Event;
+import brobot.task.Task;
+import brobot.task.TaskList;
+import brobot.task.Todo;
+
+
+/**
+ * Represents and handles the storage of the Duke Program"
+ */
+public class Storage {
+    private final String filePath;
+    private final File file;
+
+    /**
+     * Constructor.
+     *
+     * @param filePath The storage file location in the system.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.file = new File(filePath);
+        try {
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            UI.printError(e);
+        }
+    }
+
+    /**
+     * Reads the list from the storage file and translates it into a TaskList object.
+     *
+     * @return A TaskList.
+     */
+    public TaskList readList() {
+        TaskList list = new TaskList();
+        try {
+            Scanner s = new Scanner(file);
+            while (s.hasNext()) {
+                String[] arr = s.nextLine().split(" \\| ", 4);
+                String taskType = arr[0];
+                switch (taskType) {
```

Comment:
I like your usage of the switch case statement here.

Reply:
Yeah! a switch statement works great here!